lib/ssh/test/ssh_agent_mock_server.erl

```diff
@@ -0,0 +1,151 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2005-2018. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+
+%%
+
+%%% Description: ssh-agent mock
+-module(ssh_agent_mock_server).
+
+-behaviour(gen_server).
+
+-include_lib("ssh/src/ssh.hrl").
+-include_lib("ssh/src/ssh_agent.hrl").
+
+-export([respond/1, check_mktemp/1]).
+-export([start_link/1, stop/1]).
+-export([init/1, handle_info/2, handle_cast/2, handle_call/3, terminate/2]).
+
+-record(state, {socket, priv_key, pub_key, socket_path}).
+
+-define(SIG_ALG, 'ssh-rsa').
+
+start_link(PrivKeyDir) ->
+    {ok, PrivKey} =ssh_file:user_key(?SIG_ALG, [{user_dir, PrivKeyDir}]),
+
+    %% We cannot use priv_dir because unix socket paths are limited to 108 bytes.
+    SocketPath = string:chomp(os:cmd("mktemp -u")),
```

Comment:
Does this work on Windows?

Reply:
No, I don't  think windows has a `mktemp` utility. But probably also not a likely OS to have an (openssh) ssh agent, so we'll skip it.

Otherwise the `mktemp` approach is ok? We checked that temporary files are cleaned up after test runs.

Reply:
I see.  Forget my comment.

Reply:
We already check if `mktemp` is in the path and skip if it is not in there. I guess this should be fine for `windows` or do you think we still need an additional os:type check @HansN?